### PR TITLE
Pcc13

### PR DESCRIPTION
**Pull request by souhailshiv**
ATTENTION: before clicking "Create Pull Request" please submit some meta data, thanks!

Difficulty level (1-10): []
Estimated time spent (hours): []
Completed (yes/no): []
I stretched my coding skills (if yes what did you learn?): []
Other feedback (what can we improve?): []

### CODE REVIEW

**Review by pmayd — CHANGES_REQUESTED** [review]
Please one PR per challenge, you have also commited files for challenge 52

**Comment by souhailshiv** [comment]
> Please one PR per challenge, you have also commited files for challenge 52

what's do you mean I'm sorry I'm still a beginner in this website

**Comment by pmayd** [comment]
No problem!  We expect you to open a pull request for each challenge you have solved. However when you look at the files tab you have pushed two files, one in folder 13 and one in 52. So you are handing in two different challenges. One pull request should only contain files in a single folder

File: 13/souhailshiv/Highest_Rated_Movie_Directors.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+        cnt[director] += len(movies)
+        c = dict(cnt)
+    for k, v in c.items():
+        if v >= MIN_MOVIES:
+            newDict[k] = v
+    avrege = dict()
+
+    for k in newDict.keys():
+        moviescores = list()
+        for item in directors[str(k)]:
```
Comment (pmayd) [inline 13/souhailshiv/Highest_Rated_Movie_Directors.py:50]: Item is not a good name, try to be more specific.

Why do you need a str cast here? Should not b be necessary

File: 13/souhailshiv/Highest_Rated_Movie_Directors.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+            if int(item[1]) >= MIN_YEAR:
```
Comment (pmayd) [inline 13/souhailshiv/Highest_Rated_Movie_Directors.py:51]: Don't use index access here, movie is a named tuple so you can access its fields with dot notation

File: 13/souhailshiv/Highest_Rated_Movie_Directors.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+        avrege.update({k: mean(moviescores)})
+
+    return avrege
+
+
+def print_results(avrege, directors):
+    avrege = {k: v for k, v in sorted(
+        avrege.items(), key=lambda item: item[1], reverse=True)}
+    i = 0
```
Comment (pmayd) [inline 13/souhailshiv/Highest_Rated_Movie_Directors.py:62]: Don't use i explicitly, use pythons enumerate function

File: 13/souhailshiv/Highest_Rated_Movie_Directors.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    for k, v in avrege.items():
+        i = i+1
+        if i == 21:
+            break
+        print(f'\n{colored("%.2d"%i,"magenta")} {k}',
```
Comment (pmayd) [inline 13/souhailshiv/Highest_Rated_Movie_Directors.py:67]: This is hard to read, maybe a helper function for rendering directors

**Review by pmayd — APPROVED** [review]
Code is not very pythonic and hard to read, there are a lot of unnecessary type casts and for loops. You create to many dictionaries and don't use python constructs efficiently. Try to spend more time thinking about what you want to code instead of hacking a solution.

**Pull request merged**